x-pack/packages/ml/anomaly_utils/fields.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ES_FIELD_TYPES } from '@kbn/field-types';
+import { ML_JOB_AGGREGATION, KIBANA_AGGREGATION, ES_AGGREGATION } from './aggregation_types';
+import { MLCATEGORY } from './field_types';
+
+/**
+ * EVENT_RATE_FIELD_ID
+ */
+export const EVENT_RATE_FIELD_ID = '__ml_event_rate_count__';
+
+/**
+ * METRIC_AGG_TYPE
+ */
+export const METRIC_AGG_TYPE = 'metrics';
+
+/**
+ * Field id
+ */
+export type FieldId = string;
+
+/**
+ * AggId is an alias of ML_JOB_AGGREGATION
+ */
+export type AggId = ML_JOB_AGGREGATION;
+
+/**
+ * Split field can be Field or null.
+ */
+export type SplitField = Field | null;
+
+/**
+ * Field definition
+ */
+export interface Field {
+  /**
+   * The field id
+   */
+  id: FieldId;
+  /**
+   * The field name
+   */
+  name: string;
+  /**
+   * The field type is based on ES field types.
+   */
+  type: ES_FIELD_TYPES;
+  /**
+   * Flag whether the field is aggregatable.
+   */
+  aggregatable: boolean;
+  /**
+   * Flag for counter.
+   */
+  counter: boolean;
+  /**
+   * Optional array of AggId.
+   */
+  aggIds?: AggId[];
+  /**
+   * Optional array fo aggregations.
+   */
+  aggs?: Aggregation[];
+  /**
+   * Optional runtime field.
+   */
+  runtimeField?: estypes.MappingRuntimeField;
+}
+
+/**
+ * Aggregation definition.
+ */
+export interface Aggregation {
+  /**
+   * The aggregation id.
+   */
+  id: AggId;
+  /**
+   * The aggregation title.
+   */
+  title: string;
+  /**
+   * The Kibana name for the aggregation.
+   */
+  kibanaName: KIBANA_AGGREGATION | null;
+  /**
+   * The ES DSL name for the aggregation.
+   */
+  dslName: ES_AGGREGATION | null;
+  /**
+   * The metric agg type.
+   */
+  type: typeof METRIC_AGG_TYPE;
+  /**
+   * The model lot agg definition.
```

Comment:
nit: 'model plot'

Reply:
Fixed in [7a447f5](https://github.com/elastic/kibana/pull/155530/commits/7a447f56a22c6bf1ce76549677e6b375786b518f).